src/export/ExportFFmpeg.cpp

```diff
@@ -508,32 +573,37 @@ bool ExportFFmpeg::InitCodecs(AudacityProject *project)
    // This happens if user refused to resample the project
    if (mSampleRate == 0) return false;
 
-   if (mEncAudioCodecCtx->global_quality >= 0)
+   if (mEncAudioCodecCtx->GetGlobalQuality() >= 0)
    {
-      mEncAudioCodecCtx->flags |= AV_CODEC_FLAG_QSCALE;
+      mEncAudioCodecCtx->SetFlags(
+         mEncAudioCodecCtx->GetFlags() | AUDACITY_AV_CODEC_FLAG_QSCALE);
    }
-   else mEncAudioCodecCtx->global_quality = 0;
-   mEncAudioCodecCtx->global_quality = mEncAudioCodecCtx->global_quality * FF_QP2LAMBDA;
-   mEncAudioCodecCtx->sample_rate = mSampleRate;
-   mEncAudioCodecCtx->channels = mChannels;
-   mEncAudioCodecCtx->channel_layout = av_get_default_channel_layout(mChannels);
-   mEncAudioCodecCtx->time_base.num = 1;
-   mEncAudioCodecCtx->time_base.den = mEncAudioCodecCtx->sample_rate;
-   mEncAudioCodecCtx->sample_fmt = AV_SAMPLE_FMT_S16;
-   mEncAudioCodecCtx->strict_std_compliance = FF_COMPLIANCE_EXPERIMENTAL;
-
-   if (mEncAudioCodecCtx->codec_id == AV_CODEC_ID_AC3)
+   else
+   {
+      mEncAudioCodecCtx->SetGlobalQuality(0);
+   }
+
+   mEncAudioCodecCtx->SetGlobalQuality(mEncAudioCodecCtx->GetGlobalQuality() * AUDACITY_FF_QP2LAMBDA);
+   mEncAudioCodecCtx->SetSampleRate(mSampleRate);
+   mEncAudioCodecCtx->SetChannels(mChannels);
+   mEncAudioCodecCtx->SetChannelLayout(mFFmpeg->av_get_default_channel_layout(mChannels));
+   mEncAudioCodecCtx->SetTimeBase({ 1, mSampleRate });
+   mEncAudioCodecCtx->SetSampleFmt(static_cast<AVSampleFormatFwd>(AUDACITY_AV_SAMPLE_FMT_S16));
+   mEncAudioCodecCtx->SetStrictStdCompliance(
+      AUDACITY_FF_COMPLIANCE_EXPERIMENTAL);
+
+   if (mEncAudioCodecCtx->GetCodecId() == mFFmpeg->GetAVCodecID(AUDACITY_AV_CODEC_ID_AC3))
```

Comment:
One more doubt!

Is this one more place where you should use the local variable `codecID` and not the id stored in the context object?